webui/react/src/utils/loadable.ts

```diff
@@ -0,0 +1,173 @@
+ * the provided function on the contained value in the passed Loadable and then
+ * flattening the result.
+ *
+ * If any of the passed or returned Loadables is NotLoaded, the result is NotLoaded.
+ */
+const flatMap = <T, U>(l: Loadable<T>, fn: (_: T) => Loadable<U>): Loadable<U> => {
+  const res = map(l, fn);
+  if (res._tag === 'Loaded') {
+    return res.data;
+  }
+  return res;
+};
+
+/**
+ * Performs a side-effecting function if the passed Loadable is Loaded.
+ */
+const forEach = <T, U>(l: Loadable<T>, fn: (_: T) => U): void => {
+  switch (l._tag) {
+    case 'Loaded': {
+      fn(l.data);
+      return;
+    }
+    case 'NotLoaded':
+      return;
+    default:
+      exhaustive(l);
+  }
+};
+
+/**
+ * If the passed Loadable is Loaded this returns the data, otherwise
+ * it returns the default value.
+ */
+const getOrElse = <T>(def: T, l: Loadable<T>): T => {
+  switch (l._tag) {
+    case 'Loaded':
+      return l.data;
+    case 'NotLoaded':
+      return def;
+    default:
+      return exhaustive(l);
+  }
+};
+
+type MatchArgs<T, U> =
+  | {
+      Loaded: (data: T) => U;
+      NotLoaded: () => U;
+    }
+  | {
+      Loaded: (data: T) => U;
+      _: () => U;
+    }
+  | {
+      NotLoaded: () => U;
+      _: () => U;
+    };
+/**
+ * Allows you to match out the cases in the Loadable with named
+ * arguments.
+ */
+const match = <T, U>(l: Loadable<T>, cases: MatchArgs<T, U>): U => {
+  switch (l._tag) {
+    case 'Loaded':
+      return 'Loaded' in cases ? cases.Loaded(l.data) : cases._();
+    case 'NotLoaded':
+      return 'NotLoaded' in cases ? cases.NotLoaded() : cases._();
+    default:
+      return exhaustive(l);
+  }
+};
+
+/** Like `match` but without argument names */
+const quickMatch = <T, U>(l: Loadable<T>, def: U, f: (data: T) => U): U => {
+  switch (l._tag) {
+    case 'Loaded':
+      return f(l.data);
+    case 'NotLoaded':
+      return def;
+    default:
+      return exhaustive(l);
+  }
+};
+
+/** Returns true if the passed object is a Loadable */
+const isLoadable = <T, Z>(l: Loadable<T> | Z): l is Loadable<T> => {
+  return ['Loaded', 'NotLoaded', 'NotFound'].includes((l as Loadable<T>)?._tag);
+};
+
+/**
+ * Groups up all passed Loadables. NotFound takes priority over
+ * NotLoaded so all([NotLoaded, NotFound, Loaded(4)]) returns NotFound
+ */
+function all<A>(ls: [Loadable<A>]): Loadable<[A]>;
+function all<A, B>(ls: [Loadable<A>, Loadable<B>]): Loadable<[A, B]>;
+function all<A, B, C>(ls: [Loadable<A>, Loadable<B>, Loadable<C>]): Loadable<[A, B, C]>;
+function all<A, B, C, D>(
+  ls: [Loadable<A>, Loadable<B>, Loadable<C>, Loadable<D>],
+): Loadable<[A, B, C, D]>;
+function all<A, B, C, D, E>(
+  ls: [Loadable<A>, Loadable<B>, Loadable<C>, Loadable<D>, Loadable<E>],
+): Loadable<[A, B, C, D, E]>;
+function all(ls: Array<Loadable<any>>): Loadable<Array<any>> {
+  const res: any[] = [];
+  for (const l of ls) {
+    if (l._tag === 'NotLoaded') {
```

Comment:
`any` can be `unknown`, then `/* eslint-disable @typescript-eslint/no-explicit-any */` would be removed

Reply:
Thanks!